pkg/daemon/update.go

```diff
@@ -425,50 +428,58 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
+func (dn *Daemon) calculatePostConfigChangeActionFromFiles(diffFileSet []string) ([]string, error) {
+	// If a machine-config-daemon-force file is present, it means the user wants to
+	// move to desired state without additional validation. We will reboot the node in
+	// this case regardless of what MachineConfig diff is.
+	if _, err := os.Stat(constants.MachineConfigDaemonForceFile); err == nil {
+		if err := os.Remove(constants.MachineConfigDaemonForceFile); err != nil {
+			return []string{}, errors.Wrap(err, "failed to remove force validation file")
+		}
+		glog.Infof("Setting post config change action to postConfigChangeActionReboot; %s present", constants.MachineConfigDaemonForceFile)
+		return []string{postConfigChangeActionReboot}, nil
+	}
+
 	filesPostConfigChangeActionNone := []string{
 		"/etc/kubernetes/kubelet-ca.crt",
 		"/var/lib/kubelet/config.json",
 	}
+	if dn.os.IsFCOS() {
+		filesPostConfigChangeActionNone = append(filesPostConfigChangeActionNone, fcosAuthKeyPath)
+	} else {
+		filesPostConfigChangeActionNone = append(filesPostConfigChangeActionNone, nonFCOSAuthKeyPath)
+	}
+
 	filesPostConfigChangeActionReloadCrio := []string{
 		constants.ContainerRegistryConfPath,
 		GPGNoRebootPath,
 		"/etc/containers/policy.json",
 	}
 
-	actions = []string{postConfigChangeActionNone}
+	actions := []string{postConfigChangeActionNone}
 	for _, path := range diffFileSet {
 		if ctrlcommon.InSlice(path, filesPostConfigChangeActionNone) {
 			continue
 		} else if ctrlcommon.InSlice(path, filesPostConfigChangeActionReloadCrio) {
 			actions = []string{postConfigChangeActionReloadCrio}
 		} else {
-			actions = []string{postConfigChangeActionReboot}
-			return
+			return []string{postConfigChangeActionReboot}, nil
 		}
 	}
-	return
+	return actions, nil
 }
 
-func calculatePostConfigChangeAction(diff *machineConfigDiff, diffFileSet []string) ([]string, error) {
-	// If a machine-config-daemon-force file is present, it means the user wants to
-	// move to desired state without additional validation. We will reboot the node in
-	// this case regardless of what MachineConfig diff is.
-	if _, err := os.Stat(constants.MachineConfigDaemonForceFile); err == nil {
-		if err := os.Remove(constants.MachineConfigDaemonForceFile); err != nil {
-			return []string{}, errors.Wrap(err, "failed to remove force validation file")
-		}
-		glog.Infof("Setting post config change action to postConfigChangeActionReboot; %s present", constants.MachineConfigDaemonForceFile)
-		return []string{postConfigChangeActionReboot}, nil
-	}
+func (dn *Daemon) calculatePostConfigChangeActionWithMCDiff(diff *machineConfigDiff, diffFileSet []string) ([]string, error) {
+	// Note this function may only return []string{postConfigChangeActionReboot} directly,
+	// since calculatePostConfigChangeActionFromFiles may find files that require a reboot
 
+	// We don't actually have to consider ssh keys changes, which is the only section of passwd that is allowed to change
```

Comment:
can you be explicit as to why we dont have to consider them here for future readers

Reply:
ah duh, basically if mc contains a change in pwd other than ssh it'll never get to this function bc the mc isnt reconcilble. :) not sure why it took me so long to remember this.